dipy/denoise/shift_twist_convolution.pyx

```diff
@@ -0,0 +1,190 @@
+import numpy as np
+cimport numpy as cnp
+cimport cython
+
+cimport safe_openmp as openmp
+from safe_openmp cimport have_openmp
+from cython.parallel import parallel, prange, threadid
+from libc.stdlib cimport malloc, free
+
+from dipy.denoise.enhancement_kernel import EnhancementKernel
+from dipy.data import get_sphere
+from dipy.reconst.shm import sh_to_sf, sf_to_sh
+
+def convolve(odfs_sh, kernel, sh_order, test_mode=False, num_threads=None):
+    """ Perform the shift-twist convolution with the ODF data and 
+    the lookup-table of the kernel.
+
+    Parameters
+    ----------
+    odfs : array of double
+        The ODF data in spherical harmonics format
+    lut : array of double
+        The 5D lookup table
+    test_mode : boolean
+        Reduced convolution in one direction only for testing
+        
```

Comment:
`num_threads` is not documented. In particular, we need to know whether default is max threads or 1.